### PR TITLE
Angular doc-gen reboot

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -9,3 +9,4 @@ dependencies:
 test:
   override:
     - gulp lint
+    - gulp docs-test
```
Comment (IgorMinar) [inline circle.yml]: this should likely be part of travis and not circle, circle is used just for linting

**Review by IgorMinar — APPROVED** [review]
otherwise lgtm. let's merge this and we can move things later if we need to. let's keep /angular.io directory just for the app.

**Comment by IgorMinar** [comment]
one more thing - we no longer use "chore" scopes. Please use "build(aio)" instead. More info: #14098

**Comment by IgorMinar** [comment]
@petebacondarwin is this good to go? if so, please remove the cleanup and wip labels and add "pr: merge" label. thanks. I also suggest you rebase.

**Comment by petebacondarwin** [comment]
rebased and ready to merge

**Review by gkalpak — COMMENTED** [review]
Shouldn't the docs app files live under the `angular.io/` directory?

File: docs/src/app/search-worker-client.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  private _createQuery(query: string) {
+    return new Observable<QueryResults>((subscriber: Subscriber<QueryResults>) => {
+
+      // get a new identifier for this query that we can match to results
+      const id = this._queryId++;
+
+      const handleMessage = (message: MessageEvent) => {
+        console.log('IN APP:', message.data);
```
Comment (petebacondarwin) [inline docs/src/app/search-worker-client.ts]: as I said, it was  POC ... that is merged into `angular/angular` master!!

File: docs/src/app/search-worker-client.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  private _queryId = 0;
+
+  constructor(url: string, private zone: NgZone) {
+    this.worker = new Worker(url);
+    this.ready = this._waitForIndex(this.worker);
+  }
+
+  search(query: string) {
+    return Observable.of(query)
+      .switchMap(query => this.ready.then(() => query))
```
Comment (petebacondarwin) [inline docs/src/app/search-worker-client.ts]: My RxJS foo is still growing but I think you are right if you mean:

```ts
  search(query: string) {
    return Observable.fromPromise(this.ready)
            .switchMap(() => this._createQuery(query));
  }
```

**Comment by petebacondarwin** [comment]
I have fixed up @gkalpak's comments and it is now again ready for merge

**Comment by IgorMinar** [comment]
landed as 1e729d7ba..b141a227f  - I rewrote a few commit messages to follow the new guidelines (the automated check is temporarily disabled so the PR didn't fail)

**Pull request closed without merging**